README.md

```diff
@@ -42,12 +42,18 @@ $provider = new Stevenmaguire\OAuth2\Client\Provider\Salesforce([
     'redirectUri'       => 'https://example.com/callback-url'
 ]);
 
-// A getAccessTokenFromYourDataStore() example.
-$existingAccessToken = $provider->getAccessToken('password', ['username'=>'USERNAME', 'password' => 'PASSWORD']);
+// If you have a token already saved in your local data store,
+// then create a routine to pull that token from your local data storage.
+$thisAccessToken = getAccessTokenFromYourDataStore();
 
-if ($existingAccessToken->hasExpired()) {
+// If you don't have a pre-saved token, use the Salesforce OAuth API to provide a token.
+if empty($thisAccessToken) {
+    $thisAccessToken = $provider->getAccessToken('password', ['username'=>'USERNAME', 'password' => 'PASSWORD']);
+}
```

Comment:
This section has nothing to do with "Refreshing a Token" - the heading under which you are proposing this documentation to exist. 

The documentation you have proposed here is an example of obtaining a brand new token. Something that is better suited in the previous section, currently described as "Authorization Code Flow." If you think that would be useful, perhaps the previous section should be renamed to:

```
### Obtaining a new Access Token

#### Authorization Code Flow

[current documentation remains here]

#### Password Flow

[your example]
```

Keep in mind that the base package includes significantly more detailed documentation that is better to reference rather than copy/paste, creating a need to keep it in sync.

Reply:
I've updated the headers to clear that up. Thanks for the feedback. 

To be clear:  
There isn't "Password Flow" . Salesforce states that the password is required to get an access token.  That means the code using the username/password is part of obtaining a new Access Token. So there's no section called "Password Flow" but instead "Getting and/or Refreshing a Token"

The existing README.md documentation states a token exists in the programmer's local data storage (somehow, but not stated how or where that token is obtained).  That is confusing given the documentation on salesforce's site states that you use the username/login/client_id to get a token. Some say the token is good for that session only (I've not tested). 

Given that (1) the interface I was asked to look at had NO locally stored tokens and (2) the salesforce documentation specified no locally stored tokens and (3) the existing README.md file makes no mention of **how** to get a token; the  existing documentation is made clearer by having a section that talks about how to get a token from salesforce.